packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -336,6 +358,14 @@ export class Portfolio extends PortfolioBase {
     if (props.tagOptions !== undefined) {
       this.associateTagOptions(props.tagOptions);
     }
+
+    cdk.Aspects.of(this).add({
+      visit(c: IConstruct) {
+        if (c instanceof Portfolio) {
```

Comment:
Instead of `instanceof` can you try doing something like this instead?
`instanceof` has some issues and we are trying to [stop using
it](https://github.com/aws/aws-cdk/pull/20564).

```ts
cdk.Aspects.of(this).add({
  visit(c: IConstruct) {
    if (c.node.id === this.node.id) {
      ...
    };
  };
});
```
I think something like this should ensure it only runs on this construct.